iroh/src/magicsock/relay_actor.rs

```diff
@@ -807,15 +807,19 @@ impl RelayActor {
             tokio::select! {
                 biased;
                 _ = self.cancel_token.cancelled() => {
-                    trace!("shutting down");
+                    debug!("shutting down");
                     break;
                 }
-                Some(Err(err)) = self.active_relay_tasks.join_next() => {
-                    if err.is_panic() {
-                        panic!("ActiveRelayActor task panicked: {err:#?}");
-                    }
-                    if !err.is_cancelled() {
-                        error!("ActiveRelayActor failed: {err:?}");
+                Some(res) = self.active_relay_tasks.join_next() => {
+                    match res {
+                        Ok(()) => (),
+                        Err(err) if err.is_panic() => {
+                            error!("ActiveRelayActor task panicked: {err:#?}");
+                        }
+                        Err(err) if err.is_cancelled() => {
+                            error!("ActiveRelayActor cancelled: {err:#?}");
+                        }
+                        Err(err) => error!("ActiveRelayActor failed: {err:#?}"),
```

Comment:
Look, aren't real rust matching semantics nice?  The compiler told me I had a case that wasn't covered (ok, ok.  The compiler is actually wrong because task errors are always either panics or cancelled, tokio doesn't allow anything else.  But still, the compiler is right.)